src/js/controller.js

```diff
@@ -95,6 +95,10 @@ export default class Controller extends Observable {
         }
         this.triggerEvent("printProfile", filename)
     }
+	
+	onRemoveMapChip = (element) =>{
```

Comment:
I've been going back and forth on this regarding whether to use arrow syntax or not. There are subtle nuances between them, especially around 'this'. For consistency's sake, please use standard function syntax for methods

Reply:
similarly, call the argument 'payload' instead of element - you can add a comment explaining what payload is

Reply:
Please also trigger an event notifying listeners that this event took place, similar to onSearchBefore and other methods in this class